libs/SmartStore/SmartStore/Classes/Extensions/SmartStore.swift

```diff
@@ -0,0 +1,105 @@
+import Foundation
+
+/*
+ SmartStore.swift
+ SalesforceSDKCore
+ 
+ Copyright (c) 2019-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+import Combine
+
+struct Constants {
+    static let PAGE_SIZE: UInt = 65536
+}
+
+/**
+ Enum for errors
+ */
+public enum SmartStoreError: Error {
+    case error(_ error: Error?)
+    case unknown
+}
+
+/**
+ SmartStore extension
+ */
+extension SmartStore {
+    /**
+     Register a soup and returns a Result<>
+     @param withName: name of the soups
+     @param withIndexPaths: array of paths to index
+     @return a Result<Bool, SmartStoreError>
+     */
+    func registerSoup(withName soupName: String, withIndexPaths indexPaths: [String]) -> Result<Bool, SmartStoreError> {
```

Comment:
@trooper2013 @bhariharan Do we want to start introducing methods that return a Result<> instead of throwing an error? I did it for this one, but there are many more we could do it for.

Reply:
Sure, that sounds good to me!